x-pack/filebeat/input/httpjson/input.go

```diff
@@ -170,47 +253,68 @@ func (in *httpjsonInput) processHTTPRequest(ctx context.Context, client *http.Cl
 			return errors.Errorf("http request was unsuccessful with a status code %d", msg.StatusCode)
 		}
 		var m, v interface{}
+		var mm map[string]interface{}
 		err = json.Unmarshal(responseData, &m)
 		if err != nil {
+			in.log.Debugw("failed to unmarshal http.response.body", string(responseData))
 			return errors.Wrapf(err, "failed to unmarshal http.response.body")
 		}
-		switch mmap := m.(type) {
+		switch obj := m.(type) {
+		// Top level Array
+		case []interface{}:
+			mm, err = in.processEventArray(obj)
+			if err != nil {
+				return err
+			}
 		case map[string]interface{}:
 			if in.config.JSONObjects == "" {
-				ok := in.outlet.OnEvent(makeEvent(string(responseData)))
-				if !ok {
-					return errors.New("function OnEvent returned false")
+				mm, err = in.processEventArray([]interface{}{obj})
+				if err != nil {
+					return err
 				}
 			} else {
-				v, err = common.MapStr(mmap).GetValue(in.config.JSONObjects)
+				v, err = common.MapStr(mm).GetValue(in.config.JSONObjects)
 				if err != nil {
 					return err
 				}
 				switch ts := v.(type) {
 				case []interface{}:
-					for _, t := range ts {
-						switch tv := t.(type) {
-						case map[string]interface{}:
-							d, err := json.Marshal(tv)
-							if err != nil {
-								return errors.Wrapf(err, "failed to marshal json_objects_array")
-							}
-							ok := in.outlet.OnEvent(makeEvent(string(d)))
-							if !ok {
-								return errors.New("function OnEvent returned false")
-							}
-						default:
-							return errors.New("invalid json_objects_array configuration")
-						}
+					mm, err = in.processEventArray(ts)
+					if err != nil {
+						return err
 					}
 				default:
-					return errors.New("invalid json_objects_array configuration")
+					return errors.Errorf("content of %s is not a valid array", in.config.JSONObjects)
 				}
 			}
-			if in.config.Pagination != nil && in.config.Pagination.IsEnabled {
-				v, err = common.MapStr(mmap).GetValue(in.config.Pagination.IDField)
+		default:
+			in.log.Debugw("http.response.body is not valid JSON", string(responseData))
+			return errors.New("http.response.body is not valid JSON")
+		}
+
+		if mm != nil && in.config.Pagination != nil && in.config.Pagination.IsEnabled {
```

Comment:
So it's not sufficient to have defined any `pagination.XX` field in your config to get it enabled? You also have to set the `pagination.enabled: true`? If so this would be contrary to how most config in beats work.

A lot optional config objects have an `Enabled *bool` field and then a method to check if it is enabled like

```
// IsEnabled returns true if the `enable` field is set to true in the yaml.
func (p *Pagination) IsEnabled() bool {
	return p != nil && (p.Enabled == nil || *p.Enabled)
}
```